### DISCUSSION TITLE
How/where to use `com.atproto.sync.listRepos`?

### DISCUSSION THREAD

**Discussion by suppergerrie2** [bluesky-social/atproto#3221] [How to do it?] [answered]
Note that I am very new to atproto/bsky ecosystem so please correct me when I say something that does not make sense!
The [docs](https://docs.bsky.app/docs/api/com-atproto-sync-list-repos) mention that:
> This endpoint is part of the atproto repository synchronization APIs. Requests usually do not require authentication, and can be made to PDS intances or Relay instances

According to the [API Hosts and Auth](https://docs.bsky.app/docs/advanced-guides/api-directory) section the relay is `https://bsky.network` and the PDS instances are in the form of `https://<NAME>.<REGION>.host.bsky.network`. So as far as I understand I´d have to make a GET request to `https://bsky.network/xrpc/com.atproto.sync.listRepos`, but this returns:
```json
{
  "error": "Not Found"
}
```
Using `https://bsky.social` as the base URL (listed as "Entryway" in the docs) instead of  `bsky.network` does return a result as expected, but only when authenticated. But even going through all of these using the cursor it misses some repos. After some digging it seems to return the results of calling listRepos against the PDS of my account. Swapping in a PDS as base URL as in the form mentioned earlier (eg. `enoki.us-east.host.bsky.network`, a different one than my PDS, which I can access _without_ auth in contrast to `https://bsky.social` ) gives different repos, as expected if `https://bsky.social` only gives the ones from mine. 
It seems strange to me that `https://bsky.network` does not work (especially as I can find earlier discussions on here where this is used), and that `https://bsky.social` returns results for one specific PDS in quite an opaque way (you need to resolve the DIDs to figure this out as nothing in the response indicates this). A potentially slightly off-topic question, how would one find out the PDS instances available?

**Comment by yamarten** [comment, marked as answer]
The failure of the Bluesky relay is due to the update of the relay. (ref. #3036) \
Relays with this API may appear in the future, but because it is an optional feature, not all relays have it.

The entryway passes all APIs except the authentication to the PDS, so although it is not intuitive, it is the correct behavior. Authentication is required to determine which PDS to forward the request to.

The comprehensive discovery of PDS is an unresolved issue, but an informal list of volunteers can be used as a tentative approach.

https://github.com/mary-ext/atproto-scraping

**Comment by bnewbold** [comment]
This is basically a bug right now. We intend to support this endpoint at `bsky.network`, and did so previously. It isn't working at the moment because we added a proxying/fan-out service ("rainbow") which does not yet proxy this endpoint through to the true relay. We'll try to get this updated soon!
